shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -505,6 +505,30 @@ public void showSystemOverlays() {
     verify(fakeWindowInsetsController).show(WindowInsetsCompat.Type.navigationBars());
   }
 
+  @Test
+  public void verifyUpdateSystemUiOverlaysAppliesCurrentTheme() {
+    View fakeDecorView = mock(View.class);
+    Window fakeWindow = mock(Window.class);
+    when(fakeWindow.getDecorView()).thenReturn(fakeDecorView);
+    Activity fakeActivity = mock(Activity.class);
+    when(fakeActivity.getWindow()).thenReturn(fakeWindow);
+    PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
+    PlatformPlugin platformPlugin = mock(PlatformPlugin.class);
+    WindowInsetsController fakeWindowInsetsController = mock(WindowInsetsController.class);
+    when(fakeWindow.getInsetsController()).thenReturn(fakeWindowInsetsController);
+
+    SystemChromeStyle testStyle =
+        new SystemChromeStyle(
+            0XFF000000, Brightness.LIGHT, true, 0XFFC70039, Brightness.LIGHT, 0XFF006DB3, true);
+
+    platformPlugin.updateSystemUiOverlays();
+    verify(platformPlugin, never()).setSystemChromeSystemUIOverlayStyle(testStyle);
+
+    platformPlugin.setSystemChromeSystemUIOverlayStyle(testStyle);
+    platformPlugin.updateSystemUiOverlays();
+    verify(platformPlugin).setSystemChromeSystemUIOverlayStyle(testStyle);
```

Comment:
could we test for the interactions with the window itself?  

Whether `setSystemChromeSystemUIOverlayStyle` is called or not is not relevant to consumers of these APIs.

Reply:
The interaction with the window is already being tested I believe (`setSystemChromeSystemUIOverlayStyle` just calls `setSystemUiOverlayStyle` -- [example](https://github.com/flutter/engine/blob/e8397b77c5066f42b5b774534dae2d9f873ac479/shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java#L175)). Is this what you're referring to?

Reply:
right similar to that test.

This will ensure that if someone refactors `setSystemChromeSystemUIOverlayStyle`, this test will break unless the exact same Android APIs are called.